database/src/backend/heed/database.rs

```diff
@@ -125,6 +272,88 @@ impl<T: Table> DatabaseRw<T> for HeedTableRw<'_, '_, T> {
         self.db.delete(self.tx_rw, key)?;
         Ok(())
     }
+
+    #[inline]
+    fn clear(&mut self) -> Result<(), RuntimeError> {
+        Ok(self.db.clear(self.tx_rw)?)
+    }
+
+    #[inline]
+    fn retain<P>(&mut self, mut predicate: P) -> Result<(), RuntimeError>
+    where
```

Comment:
I don't really think this method will have any use, IMO this one should be removed.